demo/text-fields.html

```diff
@@ -104,32 +108,114 @@ <h3 demo-section>States</h3>
 
-          <vaadin-text-field label="Label" placeholder="Placeholder"></vaadin-text-field>
+          <vaadin-text-field label="Label and placeholder" placeholder="Placeholder"></vaadin-text-field>
 
+          <vaadin-text-field label="Label always floating" theme="always-float-label"></vaadin-text-field>
+        </div>
+      </template>
+    </nice-demo-snippet>
+
+    <h3 demo-section>Prefix and suffix</h3>
+    <nice-demo-snippet>
+      <template slot="source">
+        <div theme="dark">
+          <vaadin-text-field placeholder="Search">
+            <iron-icon slot="prefix" icon="vaadin:search"></iron-icon>
+          </vaadin-text-field>
+
+          <vaadin-text-field label="Length"
+                             theme="always-float-label"
+                             placeholder="123"
+                             pattern="[0-9]{0,3}"
+                             prevent-invalid-input
+                             style="width: 6ch;">
+            <span slot="suffix">cm</span>
+          </div>
+        </vaadin-text-field>
+
+        <custom-style>
+          <style>
+            iron-icon[slot] {
+              --iron-icon-width: 1em;
+              --iron-icon-height: 1em;
+              padding: 4px;
+              margin-right: 4px;
+            }
+          </style>
+        </custom-style>
+      </template>
+    </nice-demo-snippet>
+
+    <h3 demo-section>States</h3>
+    <nice-demo-snippet>
+      <template slot="source">
+        <div theme="dark">
           <vaadin-text-field label="Focused" value="Value" focused></vaadin-text-field>
 
+          <vaadin-text-field label="Disabled" value="Value" disabled></vaadin-text-field>
+
+          <vaadin-text-field label="Read-only" value="Value" readonly></vaadin-text-field>
+
+          <vaadin-text-field label="Disabled" placeholder="Placeholder" disabled></vaadin-text-field>
+
+          <vaadin-text-field label="Read-only" placeholder="Placeholder" readonly></vaadin-text-field>
+
+          <vaadin-text-field label="Required" placeholder="I need a value" required></vaadin-text-field>
+
           <vaadin-text-field
             label="With error message"
             value="Invalid"
             pattern="[0-9]+"
-            invalid
-            error-message="Numbers only!">
+            invalid error-message="Numbers only!">
           </vaadin-text-field>
+        </div>
+      </template>
+    </nice-demo-snippet>
 
-          <vaadin-text-field label="Disabled" value="Value" disabled></vaadin-text-field>
+    <h3 demo-section>Types</h3>
+    <nice-demo-snippet>
+      <template slot="source">
+        <div theme="dark">
+          <vaadin-text-field label="Text" value="Value"></vaadin-text-field>
+          <vaadin-password-field label="Password" value="Secret –&nbsp;hush!" placeholder="Type a secret"></vaadin-password-field>
+          <br>
+          <vaadin-text-area label="Text area" value="Value" placeholder="Placeholder"></vaadin-text-area>
```

Comment:
Looks like the Material text area doesn't grow in height as you type more lines to it. Here's how it should work: https://cdn.vaadin.com/vaadin-valo-theme/2.0.0-alpha5/demo/text-fields.html#types

Reply:
Now it doesn't grow, since Material UI guidelines define that text area should scroll vertically when the cursor reaches the bottom of the field instead of growing the area. Design decision needed if this should work as in Valo theme or as in Material component guidelines?

Reply:
Ah, didn't consider that. `<paper-textarea>` at least seems to grow indefinitely: https://www.webcomponents.org/element/PolymerElements/paper-input/demo/demo/index.html @jouni?

Reply:
Ah, right. Material design makes a distinction between a ”multi-line text-field” and a “text area”. Our text-field does not support multi-line, so I think we should cover both of those use cases with our text-area. Not 100%, but is there a similar limit to how much a ”multi-line text field” can grow?

Reply:
Points to @h00k00 for being very observant and pedantic with the Material spec, I appreciate that! In this case we should lean more towards the “inspired by”, in a way, that our `vaadin-text-area` works more like a ”multi-line text field”.

Reply:
And if a developer needs to follow the MD text area specs, they can add max-height on `vaadin-text-area` manually.

Reply:
Ok